ISM43362/ISM43362.cpp

```diff
@@ -468,18 +468,35 @@ int ISM43362::scan(WiFiAccessPoint *res, unsigned limit)
         if (res != 0 && limit != 0 && cnt >= limit) {
             /* reached end */
             if (AP_Scan_1by1 == true) {
-                /* continue to receive AP 1 by 1 from firmware until we get OK */
-                while ((strncmp("OK\r", (char *)tmp, 2) != 0)) {
-                    _parser.flush();
-                    _parser.send("MR");
-                    _parser.recv("%[^\n^\r]\r\n", tmp);
+                /* continue to receive AP 1 by 1 from firmware until we get OK or ERROR */
+                while (strncmp("OK", (char *)tmp, 2) != 0) {
+                    if (strncmp("ERROR", (char *)tmp, 5) == 0) {
+                        print_rx_buff();
+                        _parser.flush();
+                        return 0;
+                    }
+                    if (!_parser.send("MR")) {
+                        print_rx_buff();
+                        _parser.flush();
+                        return 0;
+                    }
+                    if (!_parser.recv("%[^\n^\r]\r\n", tmp)) {
+                        print_rx_buff();
+                        _parser.flush();
+                        return 0;
+                    }
                 }
             }
             break;
         }
 
         if (AP_Scan_1by1 == true) {
-            /* In case of scan AP 1 by 1, end of list is detected thanks to OK */
+            /* In case of scan AP 1 by 1, end of list is detected thanks to OK or ERROR*/
+            if (strncmp("ERROR", (char *)tmp, 5) == 0) {
```

Comment:
In case of ERROR you already returned 0 earlier so I am not sure if this check is needed